vm/compactor.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2019-2021, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package vm
+
+import (
+	"time"
+
+	log "github.com/inconshreveable/log15"
+
+	"github.com/ava-labs/spacesvm/chain"
+)
+
+func (vm *VM) compactCall(prefix *chain.PrefixRange) {
+	// Lock to prevent concurrent modification of state
+	vm.ctx.Lock.Lock()
+	defer vm.ctx.Lock.Unlock()
```

Comment:
Does this lock share with avalanchego or any other routine in spacesvm?

Reply:
This is the "global interpreter lock". AvalancheGo won't allow any function to be called on the VM interface while this is held (including RPC requests).